### DISCUSSION TITLE
[HELP] Unable to connect to Neptune Cluster

### DISCUSSION THREAD

**Discussion by KaylanHusband** [aws/graph-explorer#144] [Q&A] [answered]
I've been trying to leverage this package to visually explore a Neptune graph and Im bumping into issues.

I've created an publically accessible EC2 instance within the Neptune VPC. I can ssh into it and telnet the Neptune cluster successfully. 

For some reason I cannot synchronize the DB.

Response not ok
Attempt Credential Refresh
Master credentials available
Response not ok
Attempt Credential Refresh
Master credentials available
403
403
Response not ok
Attempt Credential Refresh
Master credentials available
Response not ok
Attempt Credential Refresh
Master credentials available
403
403

What could be causing this? Am I missing anything?

**Comment by michaelnchin** [comment]
@KaylanHusband , does your Neptune cluster have IAM Db Auth enabled?

**Reply by KaylanHusband** [reply]
sudo docker run --rm -it okigan/awscurl \
https://<neptune_endpoint>:8182/status \
--access_key *** \
--secret_key *** \
--service neptune-db

{"status":"healthy","startTime":"Thu Jul 13 16:33:01 UTC 2023","dbEngineVersion":"1.1.1.0.R7","role":"writer","dfeQueryEngine":"enabled","gremlin":{"version":"tinkerpop-3.5.2"},"sparql":{"version":"sparql-1.1"},"opencypher":{"version":"Neptune-9.0.20190305-1.0"},"labMode":{"ObjectIndex":"disabled","ReadWriteConflictDetection":"enabled"},"features":{"LookupCache":{"status":"Available"},"ResultCache":{"status":"enabled"},"IAMAuthentication":"enabled","Streams":"disabled","AuditLog":"enabled"},"settings":{"clusterQueryTimeoutInMs":"900000"}}

**Comment by KaylanHusband** [comment]
@michaelnchin 
Yes it does, and Im able to make sigv4 request through aws curl from the proxy as recommended here https://github.com/aws/graph-explorer/issues/127

I selected AWS IAM Auth Enabled

**Reply by michaelnchin** [reply]
Thanks for confirming. 

We are aware of a [bug](https://github.com/aws/graph-explorer/pull/140) currently affecting Graph Explorer connectivity to IAM enabled clusters, and have a fix pending: https://github.com/aws/graph-explorer/pull/140

We're planning to include the fix in an upcoming patch release, please stay tuned for updates.

**Reply by KaylanHusband** [reply]
@michaelnchin 
When do you think these changes will be in? Is there another version I can use in the meantime?

**Reply by michaelnchin** [reply, marked as answer]
The fixes are now merged - we will have the formal release out later this week, but you should be now be able to pull and build the patched version via the [setup documentation](https://github.com/aws/graph-explorer#steps-to-install-graph-explorer).

**Reply by KaylanHusband** [reply]
@michaelnchin 

The issue is now resolved. Thanks!

**Reply by joywa** [reply]
Thanks for the collaboration @KaylanHusband!

**Comment by KaylanHusband** [comment]
Followed the steps [here](https://github.com/aws/graph-explorer/discussions/137) and now im getting

Master credentials available
	Proxy server located at https://localhost
Response not ok
Attempt Credential Refresh
Master credentials available
Response not ok
Attempt Credential Refresh
Master credentials available
403
403
Response not ok
Attempt Credential Refresh
Master credentials available
Response not ok
Attempt Credential Refresh
Master credentials available
500
500
